test/extended/prometheus/image_pulls.go

```diff
@@ -0,0 +1,71 @@
+package prometheus
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	helper "github.com/openshift/origin/test/extended/util/prometheus"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/upgrades"
+)
+
+// ImagePullsAreFast asserts that image pulls complete quickly.
+type ImagePullsAreFast struct {
+	threshold time.Duration
+}
+
+func (t *ImagePullsAreFast) Name() string {
+	return "image-pulls-are-fast"
+}
+
+func (t *ImagePullsAreFast) DisplayName() string {
+	return "[sig-node] Image pulls are fast"
+}
+
+func (t *ImagePullsAreFast) Setup(_ *framework.Framework) {
+	if t.threshold.Seconds() == 0 {
+		t.threshold = 2 * time.Minute
+	}
+	return
+}
+
+func (t *ImagePullsAreFast) Test(f *framework.Framework, done <-chan struct{}, upgrade upgrades.UpgradeType) {
+	ctx := context.Background()
+	<-done
+
+	g.By(fmt.Sprintf("verifying that pull durations do not exceed %s", t.threshold))
+
+	oc := exutil.NewCLIWithFramework(f)
+
+	// we only consider samples since the beginning of the test
+	testDuration := exutil.DurationSinceStartInSeconds().String()
+
+	query := fmt.Sprintf(`
+sort_desc(count_over_time((
+  max(container_runtime_crio_operations_latency_microseconds{operation_type="PullImage"})
+)[%[1]s:1s]) > %d * 1e6)
+`, testDuration, int(t.threshold.Seconds()))
+	response, err := helper.RunQuery(ctx, oc.NewPrometheusClient(ctx), query)
+	o.Expect(err).NotTo(o.HaveOccurred(), "unable to retrieve CRI-O pull latencies")
+
+	if len(response.Data.Result) == 0 {
+		return
+	} else if len(response.Data.Result) > 1 {
+		framework.Failf("expected a single series, got %d: %v", len(response.Data.Result), response.Data.Result)
+	}
+
+	series := response.Data.Result[0]
+	framework.Failf("PullImage operations over %s for %s seconds", t.threshold, series.Value)
```

Comment:
Should we have this flake for now until we're sure how often it's going to hit? Would like to avoid a sudden regression in job pass rates because it turns out this happens all the time. (or are we confident it doesn't?)

Reply:
pivoted to `Flakef` with e9d6a9e0ab -> 688e4416da.